account-view/src/lib.rs

```diff
@@ -0,0 +1,873 @@
+//! Data structures to represent account information.
+
+#![no_std]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![allow(clippy::arithmetic_side_effects)]
+
+use {
+    core::{
+        marker::PhantomData,
+        mem::{size_of, ManuallyDrop},
+        ops::{Deref, DerefMut},
+        ptr::{write, write_bytes, NonNull},
+        slice::{from_raw_parts, from_raw_parts_mut},
+    },
+    solana_address::Address,
+    solana_program_error::{ProgramError, ProgramResult},
+};
+
+/// Maximum number of bytes a program may add to an account during a
+/// single top-level instruction.
+pub const MAX_PERMITTED_DATA_INCREASE: usize = 1_024 * 10;
+
+/// Value to indicate that an account is not borrowed.
+///
+/// This value is the same as `solana_program_entrypoint::NON_DUP_MARKER`.
+pub const NOT_BORROWED: u8 = u8::MAX;
+
+/// Raw account data.
+///
+/// This struct is wrapped by [`AccountView`], which provides safe access
+/// to account information. At runtime, the account's data is serialized
+/// directly after the `Account` struct in memory, with its size specified
+/// by [`Account::data_len`].
+#[repr(C)]
+#[cfg_attr(feature = "copy", derive(Copy))]
+#[derive(Clone, Default)]
+pub struct Account {
```

Comment:
There is already a public `Account` type defined in solana-account. We could name this one as `RuntimeAccount` or another name to avoid the name clash. It is not really used directly in programs, only the entrypoint uses it.